### PR TITLE
feat: add Timestamp, Reference, and GeoPoint process

### PR DESCRIPTION
**Pull request by smomin**
- add code to process Reference field type.  display reference path at this time.
- add code to process timestamp.  render full date format , e.g. 2021-05-24T06:31:32.829Z
- add code to process GeoPoint to Algolia specific format.

### CODE REVIEW

**Review by Haroenv — COMMENTED** [review]
looks good, although I don't see any new tests for geo etc. transformations, will you add those?

**Comment by smomin** [comment]
@Haroenv I have added the tests for the field type except for GeoPoint.   The cloud function test util didn't have support for it.  I will have to create an integration test.  As for the transformer, what are you referring to?

**Comment by smomin** [comment]
@Haroenv let me know if you approve this PR.  I will merge it in and then create another to bump up the version.

File: functions/src/processors.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    return ['_geoloc', processGeoPoint(value)];
+  } else if (value instanceof Timestamp) {
+    return [field, processTimestamp(value)];
+  } else if (value instanceof Array) {
+    return [field, processArray(value)];
+  } else if (value instanceof Object) {
+    return [field, processObject(value)];
+  }
+  return [field, value]
+}
```
Comment (Haroenv) [inline functions/src/processors.ts:46]: this is the code that's not tested yet :)

**Comment by Haroenv** [comment]
@smomin I think you did not yet push your updates

File: functions/src/processors.ts (hunk 1 above)
Comment (smomin) [reply functions/src/processors.ts:46]: Actually it is.  It's part of the main function that is testing each process.   My assertions to validate the expected and actual account for these new field types.

```
---------------|---------|----------|---------|---------|-------------------
File           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
---------------|---------|----------|---------|---------|-------------------
All files      |   89.51 |    76.67 |      80 |   89.44 |
 config.ts     |     100 |      100 |     100 |     100 |
 extract.ts    |   89.29 |       50 |     100 |   88.89 | 39,57,70
 index.ts      |   89.74 |       75 |     100 |   89.74 | 48,62,73,93
 logs.ts       |   82.61 |      100 |      60 |   82.61 | 35,39,43,63
 processors.ts |   89.47 |    83.33 |   77.78 |   89.47 | 37,54,67,80
 util.ts       |     100 |      100 |     100 |     100 |
 version.ts    |     100 |      100 |     100 |     100 |
---------------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       5 passed, 5 total
Snapshots:   0 total
Time:        4.313 s
```

Comment (smomin) [reply functions/src/processors.ts:46]: I spoke with the Firebase team on the GeoPoint unit test.  They will get back to me.

**Review by Haroenv — APPROVED** [review]
makes sense, let's go with this

**Pull request merged**